types/src/data_availability_header.rs

```diff
@@ -18,7 +19,6 @@ use crate::{bail_validation, Error, Result, ValidateBasic, ValidationError};
 pub struct DataAvailabilityHeader {
     pub row_roots: Vec<NamespacedHash>,
     pub column_roots: Vec<NamespacedHash>,
-    pub hash: Hash,
```

Comment:
why do we remove `hash` there?

Reply:
I mean, it has to be computed at least once during validate (which we perform on each decoding) so it will just result in additional computations when someone needs it

Reply:
There are multiple reasons:

* Since `row_roots` and `column_roots` are public, if they are modified, the `hash` gets invalidated. This was confusing when I was implementing the test utils of `ExtendedHeader`.
* I think I prefer to keep it consistent with `tendermint` types.
* After validation it can be retrieved from `extended_header.header.data_hash`, we can add `ExtendedHeader::dah_hash` method that will return the validated one. We did the same for `Header::hash()`.